tests/test_job_tracker.py

```diff
@@ -0,0 +1,385 @@
+
+        # Check each listener registration
+        calls = mock_scheduler.add_listener.call_args_list
+        assert calls[0][0][0] == job_tracker.on_job_submitted
+        assert calls[0][0][1] == EVENT_JOB_SUBMITTED
+        assert calls[1][0][0] == job_tracker.on_job_executed
+        assert calls[1][0][1] == EVENT_JOB_EXECUTED
+        assert calls[2][0][0] == job_tracker.on_job_error
+        assert calls[2][0][1] == EVENT_JOB_ERROR
+        assert calls[3][0][0] == job_tracker.on_job_missed
+        assert calls[3][0][1] == EVENT_JOB_MISSED
+        assert calls[4][0][0] == job_tracker.on_job_max_instances
+        assert calls[4][0][1] == EVENT_JOB_MAX_INSTANCES
+
+    @patch("zino.job_tracker.datetime")
+    @patch("zino.job_tracker._log")
+    def test_sigusr1_should_format_job_table_sorted_by_duration(self, mock_log, mock_datetime):
+        """Test the formatted output of SIGUSR1 handler is sorted by duration."""
+        job_tracker = JobTracker()
+        current_time = datetime(2024, 1, 1, 12, 0, 0)
+        mock_datetime.now.return_value = current_time
+
+        # Add jobs with varying durations
+        job_tracker.running_jobs = {
+            "job1": {
+                "start_time": current_time - timedelta(days=1, hours=2, minutes=30, seconds=45.123),
+                "job_name": "LongRunningTask",
+            },
+            "job2": {
+                "start_time": current_time - timedelta(minutes=5, seconds=30.5),
+                "job_name": "QuickTask",
+            },
+            "job3": {
+                "start_time": current_time - timedelta(hours=1),
+                "job_name": None,
+            },
+        }
+
+        job_tracker._handle_sigusr1()
+
+        # Verify the output includes all jobs and is properly formatted
+        info_calls = [str(call) for call in mock_log.info.call_args_list]
+        output = "\n".join(info_calls)
+
+        assert "Total running jobs: 3" in output
+        assert "Job ID" in output
+        assert "Duration" in output
+        assert "Started" in output
+        assert "LongRunningTask" in output
+        assert "QuickTask" in output
+
+        # Verify jobs are sorted by duration (longest first)
+        # Find the positions of each job in the output
+        longrunning_pos = output.find("LongRunningTask")
+        job3_pos = output.find("job3")
+        quicktask_pos = output.find("QuickTask")
+
+        # All jobs should be present
+        assert longrunning_pos != -1, "LongRunningTask not found in output"
+        assert job3_pos != -1, "job3 not found in output"
+        assert quicktask_pos != -1, "QuickTask not found in output"
+
+        # Verify order: LongRunningTask (1d 2h) should come before job3 (1h) which should come before QuickTask (5m)
+        assert longrunning_pos < job3_pos, "LongRunningTask should appear before job3 (sorted by duration)"
+        assert job3_pos < quicktask_pos, "job3 should appear before QuickTask (sorted by duration)"
```

Comment:
Why are the two tests that are checking for how the result of the job table looks like so far apart? I feel like they should be together